src/coreclr/System.Private.CoreLib/src/System/Runtime/DependentHandle.cs

```diff
@@ -0,0 +1,259 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+#if !DEBUG
+using Internal.Runtime.CompilerServices;
+#endif
+
+namespace System.Runtime
+{
+    /// <summary>
+    /// Represents a dependent GC handle, which will conditionally keep a dependent object instance alive
+    /// as long as a target object instance is alive as well, without representing a strong reference to the
+    /// target object instance. That is, a <see cref="DependentHandle"/> value with a given object instance as
+    /// target will not cause the target to be kept alive if there are no other strong references to it, but
+    /// it will do so for the dependent object instance as long as the target is alive.
+    /// <para>
+    /// This type is conceptually equivalent to having a weak reference to a given target object instance A, with
+    /// that object having a field or property (or some other strong reference) to a dependent object instance B.
+    /// </para>
+    /// </summary>
```

Comment:
The details are nice, but this is way too long for a summary, which should be at most one sentence (it's what pops up in IntelliSense for a method).  Can you move everything but the first sentence to remarks, editing it appropriately?  Thanks!

Reply:
Fixed this in https://github.com/dotnet/runtime/pull/54246/commits/c463d54a640ad20e5e23859c2a04ca59757b8ffe, as well as all the other review comments below 🙂